### ISSUE TITLE
Possible Enhancement: Callback/Notification when EVCloudKitDao is finished initializing

### ISSUE THREAD

**Issue by tbaggett** [evermeer/EVCloudKitDao#25]
It is currently possible to use EVCloudKitDao's functionality before it has completed initialization, specifically before the completion closure being called by accountStatusWithCompletionHandler() and the user's account status being assigned to the accountStatus property. While the accountStatus property can be checked for a nil value, I don't see an obvious way to receive a callback or notification when initialization has completed without making source changes to the project.

I thought about calling the accountStatusWithCompletionHandler method directly, but I doubt that getting a callback from my own execution will guarantee that EVCloudKitDao's call has also been completed.

I need this due to the new Apple TV's heavy dependence on iCloud for application data storage. The first view displayed by my app depends on the user's logged in status and if they've previously created documents with my app. I need to display an error screen if the user isn't (a) logged in to their iCloud account or (b) they are unable to save documents using the account, or a loading screen until EVCloudKitDao is finished initializing and I've checked for previously created documents.

I am admittedly new to both Swift and (i/tv)OS(x) development, so I won't be surprised if there is a simple way to monitor the initialization state without modifying the project code. Do you have any suggestions on how this can be handled? Thanks.

**Comment by evermeer** [comment]
You are right. The initialization is calling the accountStatusWithCompletionHandler asynchronously. Since it's a method that (always?) finishes instant I did not noticed a potential problem here. The easiest fix here is to make this call synchronous don't you think?

I think in your case your first view should be some sort of a splash screen with something like a spinner and a message 'logging in'. Then call the function discoverUserInfo and in the completion handler navigate to your actual app screen. When there is a completion handler, you could show an account status message. This is how I do it in my app.

**Comment by evermeer** [comment]
I just pushed version 2.11.2 to GitHub
In this version the initialization is now synchronous.
So you will always have an accountStatus

I assumed that the accountStatusWithCompletionHandler will never be a long lasting operation. 
If I'm wrong, then it's better to wrap the first call to EVCloudKitDao (probably the discoverUserInfo) in an Async block.

**Comment by tbaggett** [comment]
Thanks as always for the lightning-fast response! I will grab the update and wrap the first call as suggested. I had added a "logging in" screen with an activity indicator, which is what prompted me to find an asynchronous way to wait on the initialization.

In my experience, the accountStatusWithCompletionHandler call has been quick to return the account status, but it wasn't always ready the first time I attempted to check it. I also assumed it would always be a quick operation, so I initially solved the issue with the code below. I don't think I ever saw more than one "Awaiting return..." log statement, but it was just enough that I had to guard against the nil condition.

    // Check user's current iCloud account status. iCloud is required to store user-created content
    var attempt: Int = 0
    while (EVCloudKitDao.privateDB.accountStatus == nil && attempt++ < 10) {
      EVLog("Awaiting return of iCloud account status")
      usleep(100000)
    }
